src/Core/Core.js

```diff
@@ -11817,14 +11817,36 @@ export default class Core {
     //   }
     // })
 
+    let pause100 = function () {
+      return new Promise((resolve, reject) => {
+        setTimeout(resolve, 100);
+      });
+    };
+
+    if (this.savingPageState) {
+      for (let i = 0; i < 100; i++) {
+        await pause100();
+
+        if (!this.savingPageState) {
+          break;
+        }
+      }
```

Comment:
Is there not an event message we can listen for?

Reply:
Decided this is something to revisit later.